server/src/main/java/org/apache/druid/initialization/ServerInjectorBuilder.java

```diff
@@ -106,7 +106,7 @@ public Injector build()
     CoreInjectorBuilder coreBuilder = new CoreInjectorBuilder(childInjector, nodeRoles).forServer();
 
     // Override with the per-service modules.
-    ServiceInjectorBuilder serviceBuilder = new ServiceInjectorBuilder(coreBuilder).addAll(
+    ServiceInjectorBuilder serviceBuilder = (ServiceInjectorBuilder) new ServiceInjectorBuilder(coreBuilder).addAll(
```

Comment:
unfortunately its needed because:
* [`ServiceInjectorBuilder extends DruidInjectorBuilder`](https://github.com/apache/druid/blob/c27f5bf52f6690e76e39381f57683b374787b222/server/src/main/java/org/apache/druid/initialization/ServiceInjectorBuilder.java#L33)
* `addAll` is defined in `DruidInjectorBuilder` to [return `DruidInjectorBuilder`](https://github.com/apache/druid/blob/c27f5bf52f6690e76e39381f57683b374787b222/server/src/main/java/org/apache/druid/guice/DruidInjectorBuilder.java#L120C28-L120C28); but returns `this` ; which due to extends will be a `ServiceInjectorBuilder`
* the constructor of `ExtensionInjectorBuilder` wants a `ServiceInjectorBuilder` 

this can be fixed - but I didn't wanted to do it in this PR